src/reformat_libyuv.c

```diff
@@ -514,139 +665,57 @@ avifResult avifImageYUVToRGBLibYUV8bpc(const avifImage * image,
-                                                int,
-                                                const uint8_t *,
-                                                int,
-                                                uint8_t *,
-                                                int,
-                                                const struct YuvConstants *,
-                                                int,
-                                                int,
-                                                enum FilterMode);
-            YUVToRGBMatrixFilter lutYuvToRgbMatrixFilter[AVIF_RGB_FORMAT_COUNT][AVIF_PIXEL_FORMAT_COUNT] = {
-                // { NONE, YUV444, YUV422, YUV420, YUV400 }                           // AVIF_RGB_FORMAT_
-                { NULL, NULL, I422ToRGB24MatrixFilter, I420ToRGB24MatrixFilter, NULL }, // RGB
-                { NULL, NULL, I422ToARGBMatrixFilter, I420ToARGBMatrixFilter, NULL },   // RGBA
-                { NULL, NULL, NULL, NULL, NULL },                                       // ARGB
-                { NULL, NULL, I422ToRGB24MatrixFilter, I420ToRGB24MatrixFilter, NULL }, // BGR
-                { NULL, NULL, I422ToARGBMatrixFilter, I420ToARGBMatrixFilter, NULL },   // BGRA
-                { NULL, NULL, NULL, NULL, NULL },                                       // ABGR
-                { NULL, NULL, NULL, NULL, NULL },                                       // RGB_565
-            };
-            YUVToRGBMatrixFilter yuvToRgbMatrixFilter = lutYuvToRgbMatrixFilter[rgb->format][image->yuvFormat];
-            if (yuvToRgbMatrixFilter != NULL) {
-                const enum FilterMode filter = ((rgb->chromaUpsampling == AVIF_CHROMA_UPSAMPLING_FASTEST) ||
-                                                (rgb->chromaUpsampling == AVIF_CHROMA_UPSAMPLING_NEAREST))
-                                                   ? kFilterNone
-                                                   : kFilterBilinear;
-                if (yuvToRgbMatrixFilter(image->yuvPlanes[AVIF_CHAN_Y],
-                                         image->yuvRowBytes[AVIF_CHAN_Y],
-                                         image->yuvPlanes[uPlaneIndex],
-                                         image->yuvRowBytes[uPlaneIndex],
-                                         image->yuvPlanes[vPlaneIndex],
-                                         image->yuvRowBytes[vPlaneIndex],
-                                         rgb->pixels,
-                                         rgb->rowBytes,
-                                         matrix,
-                                         image->width,
-                                         image->height,
-                                         filter) != 0) {
-                    return AVIF_RESULT_REFORMAT_FAILED;
-                }
-                return AVIF_RESULT_OK;
-            }
+static avifResult avifImageDownscaleTo8bpcYUVAndConvertToRGBLibYUV(const avifImage * image,
```

Comment:
Optional: Maybe change "Downscale" to "Downshift"?

"Downscale" could also mean reducing the size of the image.

Reply:
or Downsample might work

Reply:
Changed it to downsample.

Reply:
"Downsample" and "subsample" mean the same thing. It means reducing sampling rate, e.g., chroma subsampling.

Reply:
Ack. I updated it to "downshift".